addons/l10n_tr_nilvera/models/res_company.py

```diff
@@ -0,0 +1,55 @@
+from odoo import fields, models
+from odoo.addons.l10n_tr_nilvera.lib.nilvera_client import NilveraClient
+
+
+class ResCompany(models.Model):
+    _inherit = 'res.company'
+
+    l10n_tr_nilvera_api_key = fields.Char(string='Nilvera API KEY', groups='base.group_system')
+    l10n_tr_nilvera_environment = fields.Selection(
+        string="Nilvera Environment",
+        selection=[
+            ('sandbox', 'Sandbox'),
+            ('production', 'Production'),
+        ],
+        required=True,
+        default='sandbox',
+    )
+    l10n_tr_nilvera_purchase_journal_id = fields.Many2one(
+        comodel_name='account.journal',
+        string='Nilvera Purchase Journal',
+        domain=[('type', '=', 'purchase')],
+        store=True,
+        compute='_compute_l10n_tr_nilvera_purchase_journal_id',
+        inverse='_inverse_l10n_tr_nilvera_purchase_journal_id',
+    )
+
+    def _compute_l10n_tr_nilvera_purchase_journal_id(self):
+        for company in self:
+            if not company.l10n_tr_nilvera_purchase_journal_id:
+                company.l10n_tr_nilvera_purchase_journal_id = self.env['account.journal'].search([
```

Comment:
Florian just told me that we could do the search outside the loop with no limit to get all the journal and then in the loop use the filtered_domain function on the record set with the _check_company_domain
It will do only one sql query to search the journals :smile: